node/src/tasks/msp_move_bucket.rs

```diff
@@ -88,230 +91,560 @@ where
-                    bucket_id: event.bucket_id,
-                    response: BucketMoveRequestResponse::Rejected,
-                },
-            );
-
-            self.storage_hub_handler
-                .blockchain
-                .send_extrinsic(call, Tip::from(0))
-                .await?
-                .with_timeout(Duration::from_secs(
-                    self.storage_hub_handler
-                        .provider_config
-                        .extrinsic_retry_timeout,
-                ))
-                .watch_for_success(&self.storage_hub_handler.blockchain)
-                .await?;
-
-            return Ok(());
+            return self.reject_bucket_move(event.bucket_id).await;
         };
 
-        // TODO: check that we have enough space to accept the bucket and reject if not (+test)
-        // TODO: check that are know all the BSPs for the files in the bucket and reject if not (+test)
-
-        let call = storage_hub_runtime::RuntimeCall::FileSystem(
-            pallet_file_system::Call::msp_respond_move_bucket_request {
-                bucket_id: event.bucket_id,
-                response: BucketMoveRequestResponse::Accepted,
-            },
-        );
-
-        info!(
-            target: LOG_TARGET,
-            "MSP: accepting move bucket request for bucket {:?}",
-            event.bucket_id,
-        );
-
-        self.storage_hub_handler
-            .blockchain
-            .send_extrinsic(call, Tip::from(0))
-            .await?
-            .with_timeout(Duration::from_secs(
-                self.storage_hub_handler
-                    .provider_config
-                    .extrinsic_retry_timeout,
-            ))
-            .watch_for_success(&self.storage_hub_handler.blockchain)
-            .await?;
-
+        let mut indexer_connection = match indexer_db_pool.get().await {
+            Ok(connection) => connection,
+            Err(error) => {
+                error!(target: LOG_TARGET, "Failed to get indexer connection after timeout: {:?}", error);
+                return self.reject_bucket_move(event.bucket_id).await;
+            }
+        };
         let bucket = event.bucket_id.as_ref().to_vec();
 
-        let mut indexer_connection = indexer_db_pool.get().await?;
-
         let forest_storage = self
             .storage_hub_handler
             .forest_storage_handler
-            .get_or_create(&event.bucket_id.as_ref().to_vec())
+            .get_or_create(&bucket)
             .await;
 
-        // TODO(improvement): Parallelize this.
-        for file in shc_indexer_db::models::File::get_by_onchain_bucket_id(
+        self.pending_bucket_id = Some(event.bucket_id);
+
+        let files = shc_indexer_db::models::File::get_by_onchain_bucket_id(
             &mut indexer_connection,
             bucket.clone(),
         )
-        .await?
+        .await?;
+
+        let total_size: u64 = files.iter().map(|file| file.size as u64).sum();
```

Comment:
We could just use the `size` field stored for each bucket to know the total size of data stored inside of it.

https://github.com/Moonsong-Labs/storage-hub/blob/0ac5d27ce44448e016e3fe5c96eb2fa564884cbd/pallets/providers/src/types.rs#L201